### DISCUSSION TITLE
odroid install: "is not a supported wheel on this platform."

### DISCUSSION THREAD

**Discussion by kro8546** [mcguirepr89/BirdNET-Pi#775] [Q&A] [answered]
Hello,

Because of the shortage of PI 4 hardware, I want to install the software on a ODROID device. 
It was installing quite good, with some modifications. But now I run into a problem. Maybe anyone can help?

In the script somewhere the line is:
 pip3 install -U -r /root/BirdNET-Pi/requirements.txt
and the failure is:
tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl is not a supported wheel on this platform.

![image](https://user-images.githubusercontent.com/68680396/222903309-cc8877c1-af17-4ec0-ae6e-c282589271b3.png)

**Comment by matthew73210** [comment, marked as answer]
What os are you running ? It works on armbian.

**Reply by kro8546** [reply]
I used Linux odroid-buster64 4.9.312-arm64.

**Reply by kro8546** [reply]
Thanks for suggestion. I will try armbian first. Maybe it requires less modification.

**Reply by kro8546** [reply]
It works on armbian Bullseye. 
Jammy is not yet working.

**Reply by matthew73210** [reply]
Ah yes I forgot to mention the specific version, sorry for the hassle.

**Comment by ehpersonal38** [comment]
Can you please copy and paste the output of the following commands to check your Python version?

```
~/BirdNET-Pi/birdnet/bin/python3
```
Then once you're in the Python shell do these commands
```
import platform
platform.architecture()
```

**Reply by kro8546** [reply]
Thanks. The output of the first command:

Python 3.7.3 (default, Oct 31 2022, 14:04:00)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.

And for the second:
('64bit', '')

**Reply by ehpersonal38** [reply]
So I think the issue is that your Python is the wrong version. You're using 3.7.3 when you should be on at least 3.9.

```
tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl
‎‎‎‎‎‎‎‎‎‎‎            ‎‎‎           ^^^ the 39 means it only works with Python 3.9.
```

My Pi currently is running Python 3.9.2 so you should try installing that version and see if it helps.

**Reply by ehpersonal38** [reply]
Did you see the solution above ^?

**Reply by kro8546** [reply]
yes I did. Thank you. Because of the old Linux version, I am trying armbian first. 
With some adjustments, the script is running now. That's good. But I have now some other problems.

**Comment by kro8546** [comment]
Suggest to read [https://github.com/PINTO0309/TensorflowLite-bin](https://github.com/PINTO0309/TensorflowLite-bin) in order to fix the problem.

**Reply by JohnTrischan** [reply]
Hey @ehpersonal38, I am running into the same issue as above. But it looks like my RaspPi 4 is running Python 3.9.2 (32-bit) Is there a way to change from 32 to 64? Fairly certain I used the Bullseye legacy 64-bit lite OS. I would assume that would give me the 64-bit python, but I am also very new to this. 

++ '[' -n /usr/bin/bash -o -n '' ']'
++ hash -r
+ pip3 install -U -r /home/triscuits/BirdNET-Pi/requirements.txt
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
ERROR: tflite_runtime-2.6.0-cp39-none-linux_aarch64.whl is not a supported wheel on this platform.
The installation exited unsuccessfully.

Python:
~/BirdNET-Pi/birdnet/bin/python3
Python 3.9.2 (default, Mar 12 2021, 04:06:34)
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> platform.architecture()
('32bit', 'ELF')
